sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -328,15 +328,17 @@ class SessionCatalog(
 
   def validateTableLocation(table: CatalogTable): Unit = {
     // SPARK-19724: the default location of a managed table should be non-existent or empty.
-    if (table.tableType == CatalogTableType.MANAGED) {
-      val tableLocation =
-        new Path(table.storage.locationUri.getOrElse(defaultTablePath(table.identifier)))
-      val fs = tableLocation.getFileSystem(hadoopConf)
-
-      if (fs.exists(tableLocation) && fs.listStatus(tableLocation).nonEmpty) {
-        throw new AnalysisException(s"Can not create the managed table('${table.identifier}')" +
-          s". The associated location('${tableLocation.toString}') already exists.")
-      }
+    // SPARK-31298: when create external table also should be a non-existent or empty
```

Comment:
This looks wrong, @AngersZhuuuu . `EXTERNAL TABLE` is used to mount an exiting table location.